interp/src/primitives/stateful.rs

```diff
@@ -384,6 +296,7 @@ impl StdReg {
             width,
             data: [Value::new(width as usize)],
             update: None,
+            cycle_count: 0,
```

Comment:
Seems to me that it would be better to remember both the input and the write_en signal, rather than this cycle_count variable. The done signal is output from the do_tick corresponds directly to whether the write_en signal was high, i.e., if the do_tick runs with a high write_en then the done signal is also high and if low it is low

Reply:
Oh yea that makes sense! The `cycle_count` thing was just how I thought of the ticks, but what you suggest is more straightforward.